woocommerce/woocommerce-ios
# [Products] Show disclosure indicators in product detail rows when row is actionable

Closes: #8745


## Description

This fixes a bug where disclosure indicators did not appear on all rows in the product details screen, even when the product details were editable.

### Changes

In https://github.com/woocommerce/woocommerce-ios/pull/7095 the decision to show the disclosure indicator in `ImageAndTitleAndTextTableViewCell` was decoupled from the `inActionable` property in the cell view model. However, the product details screen wasn't updated, so the disclosure indicator wasn't shown even for actionable rows using that cell type.

This updates `ProductFormTableViewDataSource` (where the view model is created for those rows) to show the disclosure indicator when the row is actionable.

## Testing instructions


1. Go to the Products tab and select a product.
2. Confirm disclosure indicators appear on the expected rows (where the row is editable and tapping it opens a new screen).
3. Go back and select a variable product.
4. Open the Variations section and select a variation.
5. Confirm disclosure indicators appear on the expected rows (where the row is editable and tapping it opens a new screen).

You can also check that disclosure indicators do not appear when they shouldn't:

1. Go to the Orders list and select an order.
2. Select an item in the order to open read-only product details.
3. Confirm disclosure indicators **don't** appear on non-selectable rows.

## Screenshots


/|Before|After
-|-|-
Product|![Simulator Screen Shot - iPhone 14 Pro - 2023-02-24 at 11 37 39](https://user-images.githubusercontent.com/8658164/221170143-9c70e06c-fb42-479e-a176-a6d25df1aef4.png)|![Simulator Screen Shot - iPhone 14 Pro - 2023-02-24 at 11 36 05](https://user-images.githubusercontent.com/8658164/221170169-59f4ac9f-d962-4c06-912e-2d2559ff6f0e.png)
Product variation|![Simulator Screen Shot - iPhone 14 Pro - 2023-02-24 at 11 38 00](https://user-images.githubusercontent.com/8658164/221170157-2aa89b4b-e542-4c0a-8288-bde5f0d2a80d.png)|![Simulator Screen Shot - iPhone 14 Pro - 2023-02-24 at 11 36 11](https://user-images.githubusercontent.com/8658164/221170184-17a422b3-27d5-4156-8881-740fdb4c8f13.png)

---
